### ISSUE TITLE
torch.utils.ffi test is not being triggered and has bitrotted

### ISSUE THREAD

**Issue by ezyang** [pytorch/pytorch#6978]
In `test_utils.py` we have

```
try:
    import cffi
    from torch.utils.ffi import compile_extension
    HAS_CFFI = True
except ImportError:
    HAS_CFFI = False
```

but `compile_extension` no longer exists, it's called `create_extension` now. So we always skip.

I applied this patch:

```
diff --git a/test/test_utils.py b/test/test_utils.py
index 9e9d358..ba9f709 100644
--- a/test/test_utils.py
+++ b/test/test_utils.py
@@ -28,11 +28,14 @@ from common import TestCase, run_tests, download_file
 
 try:
     import cffi
-    from torch.utils.ffi import compile_extension
     HAS_CFFI = True
 except ImportError:
+    raise
     HAS_CFFI = False
 
+if HAS_CFFI:
+    from torch.utils.ffi import create_extension
+
 
 class SimplePlugin(Plugin):
 
@@ -352,20 +355,22 @@ class TestFFI(TestCase):
         self.tmpdir = tempfile.mkdtemp()
         os.chdir(self.tmpdir)
         sys.path.append(self.tmpdir)
+        print(self.tmpdir)
 
     def tearDown(self):
-        shutil.rmtree(self.tmpdir)
+        #shutil.rmtree(self.tmpdir)
+        pass
 
     @unittest.skipIf(not HAS_CFFI, "ffi tests require cffi package")
     def test_cpu(self):
-        compile_extension(
+        create_extension(
             name='test_extensions.cpulib',
-            header=test_dir + '/ffi/src/cpu/lib.h',
+            headers=[test_dir + '/ffi/src/cpu/lib.h'],
             sources=[
                 test_dir + '/ffi/src/cpu/lib1.c',
                 test_dir + '/ffi/src/cpu/lib2.c',
             ],
-            verbose=False,
+            verbose=True,
         )
         from test_extensions import cpulib
         tensor = torch.ones(2, 2).float()
@@ -386,9 +391,9 @@ class TestFFI(TestCase):
 
     @unittest.skipIf(not HAS_CFFI or not HAS_CUDA, "ffi tests require cffi package")
     def test_gpu(self):
-        compile_extension(
+        create_extension(
             name='gpulib',
-            header=test_dir + '/ffi/src/cuda/cudalib.h',
+            headers=[test_dir + '/ffi/src/cuda/cudalib.h'],
             sources=[
                 test_dir + '/ffi/src/cuda/cudalib.c',
             ],
```

But no compiling seems to happen when I run it, and so I get this error:

```
======================================================================
ERROR: test_cpu (__main__.TestFFI)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test/test_utils.py", line 375, in test_cpu
    from test_extensions import cpulib
  File "/tmp/tmp035br9l2/test_extensions/cpulib/__init__.py", line 3, in <module>
    from ._cpulib import lib as _lib, ffi as _ffi
ModuleNotFoundError: No module named 'test_extensions.cpulib._cpulib'


```

I haven't looked into how `torch.utils.ffi` is actually implemented yet.

CC @soumith @apaszke

**Comment by ezyang** [comment]
OK, looks like I just have to call `build()` on the return of `create_extension`

**Comment by ezyang** [comment]
Fixes in #6913, but Windows support looks like it will need some more love.
